e2e/run.js

```diff
@@ -0,0 +1,89 @@
+const { spawn, spawnSync } = require('child_process')
+const db = require('./db')
+
+const debug = false
+const createPostgresContainer = true
+const deletePostgresContainer = true
+
+const postgresContainerName = 'indexer-postgres-e2e'
+
+function run(command, args, options) {
+  options = options || {}
+  if (debug) console.log(`${command} \\\n  ${args.join(' \\\n  ')}`)
+  const result = spawnSync(command, args, options)
+  if (result.error) {
+    throw result.error
+  } else if (result.status) {
+    throw new Error(`${command} failed: ${result.status}`)
+  } else if (result.signal) {
+    throw new Error(`${command} exited from signal: ${result.signal}`)
+  }
+  return result
+}
+
+function docker(args, options) {
+  options = options || {}
+  options = {
+    ...options,
+    env: {
+      ...process.env,
+      ...options.env
+    }
+  }
+  return run('docker', args, options)
+}
+
+function sleep(milliseconds) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, milliseconds)
+  })
+}
+
+async function main() {
+  let indexerChildProcess = null
+
+  const exitHandler = () => {
+    if (indexerChildProcess) {
+      indexerChildProcess.kill('SIGINT')
+    }
+    if (deletePostgresContainer) {
+      docker(['rm', '-f', postgresContainerName])
+    }
+  }
+  process.on('exit', exitHandler)
+  process.on('SIGINT', () => process.exit(1))
+  process.on('SIGTERM', () => process.exit(1))  
+  
+  if (createPostgresContainer) {
+    docker(['run',
+            '--name', postgresContainerName,
+            '--rm',
+            '-d',
+            '-p', '5432:5432',
+            '-e', `POSTGRES_DB=${db.postgresDatabase}`,
+            '-e', `POSTGRES_PASSWORD=${db.postgresPassword}`,
+            'postgres'])
+  }
+
+  // Ensure DB is up before starting indexer
+  await db.psql('SELECT 1;', 30)
+  
+  indexerChildProcess = spawn('node', ['./dist/bin/indexer.js'], {
+    stdio: 'inherit',
+    env: {
+      WEB3_PROVIDER_URL: 'https://alfajores-forno.celo-testnet.org',
+      ...process.env,
+    }
+  })
+  console.log('Waiting for DB to have some contents...')
+  const rows = await db.psql('SELECT * FROM transfers LIMIT 1;', 30)
```

Comment:
Eventually when we have more or more complex tests it would be good to have them in a separate file as well, but for now this looks great 😄